compiler/rustc_resolve/src/macros.rs

```diff
@@ -443,11 +443,22 @@ impl<'a> ResolverExpand for Resolver<'a> {
                 PathResult::NonModule(partial_res) if partial_res.unresolved_segments() == 0 => {
                     return Ok(true);
                 }
+                PathResult::NonModule(..) |
+                // HACK(Urgau): This shouldn't be necessary
+                PathResult::Failed { is_error_from_last_segment: false, .. } => {
```

Comment:
This is kind of throwing the baby out with the bathwater.
But if you don't want to modify `fn resolve_path_with_ribs`, then it's the best that can be done, I guess.

Reply:
Quite the contrary I want to modify it. Didn't you saw https://github.com/rust-lang/rust/pull/97391#discussion_r881976616 ?